src/components/loading/Loader.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { Suspense } from "react";
+import { CircularLoader } from "@dhis2/ui";
+import styles from "./Loader.module.css";
+import { QueryResponse } from "../../types/query";
+import { NoticeBox } from "@dhis2/ui";
+import i18n from "@dhis2/d2-i18n";
+
+interface LoaderProps {
+    children: React.ReactNode;
+    queryResponse: QueryResponse;
+    label?: string;
+}
+export const Loader = ({ children, queryResponse, label }: LoaderProps) => {
```

Comment:
`<Loader />` needs a `@TODO` comment with a linked Jira issue to get feedback from @cooper-joe so we know how we want to display error and loading states.

Also this component is, right now, only used in the `src/app` component, so it should only be there. `src/components` is - similarly to `src/lib` - for shared components. Until we've gotten feedback from @cooper-joe about how to deal with loading and error states, I'm against making asumptions & adding concretions in shared folders that are not being shared. As soon as we know that this component could/should be used elsewhere, that's when I'd move it to the `components` folder

Reply:
I feel like this is such a generic component that we can assume it will be shared, so I disagree there. I do agree we can talk to Joe though.

Reply:
Fair enough. I suppose this is also simple enough to be changed very easily if that assumption turns out to be wrong

Reply:
If this is the _first load_ (described [here](https://docs.google.com/document/d/1FJ5NZm1vLCxhB-LXOApKkM5iNDbWkgrYIpSn5lHu97c/edit#heading=h.r5vcxyx16192)) then I think it's OK to assume that it'll be generic. Things get a bit more complex with loading between pages (e.g. loading inside the data-table), but for "load the object list view", we can assume a generic component I think.